salinity/salinity_front/views.py

```diff
@@ -8,26 +8,42 @@
 import logging
 import collections
 from salinity_front.models import CheckRedis
+import jsonpickle
+from time import time
+import threading
+
+server_con = CheckRedis("localhost")
+roles = {'app':['web', 'lb', 'php', 'app', 'util', 'queue', 'solr', 'es', 'node', 'nfs', 'sftp', 'rsyslog', 'mmonit'], 'ci': ['slave', 'ci', 'tarantula', 'awsutils']}
+no_stg = ['rsyslog', 'mmonit']
+envs = {'app':['qa', 'stg', 'prd'], 'ci':['ci']}
+
+def update_redis_context():
+    while(True):
+        context_dict = {}
+        for env_type, env_list in envs.iteritems():
+            for env in env_list:
+                for role in roles[env_type]:
+                    context_dict[role + "_" + env] = {'status':server_con.check_failed_role(role, env), 'role':role, 'env':env}
+        redis_context = jsonpickle.encode(context_dict)
+        server_con.write_context(redis_context)
```

Comment:
other things is cosmetic:    

if you ever what to test code that executes in ``update_redis_context()``  it would be hard to do because if the while(True).     However it would be possible if
  ``def update_redis_context():
       while(True):
           update_redis():
           # all code here
           #    this way  we can test   update_redis() if we want to.